tests/test_admin.py

```diff
@@ -248,6 +261,64 @@ def test_advanced_settings_link(self):
         )
 
 
+class SetHomeViewTestCase(CMSTestCase):
+    def test_get_method_is_not_allowed(self):
+        pagecontent = PageContentWithVersionFactory()
+        with self.login_user_context(self.get_superuser()):
+            response = self.client.get(
+                self.get_admin_url(PageContent, "set_home_content", pagecontent.pk)
+            )
+        self.assertEqual(response.status_code, 405)
+
+    def test_root_page_is_allowed_to_set_home(self):
+        version = PageVersionFactory(content__page__node__depth=1)
+        pagecontent = version.content
+        version.publish(self.get_superuser())
+        with self.login_user_context(self.get_superuser()):
+            response = self.client.post(
+                self.get_admin_url(PageContent, "set_home_content", pagecontent.pk)
+            )
+        url = reverse("admin:cms_pagecontent_changelist")
+
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(response.url, url)
+
+    def test_non_root_page_should_not_allowed_to_set_home(self):
+        version = PageVersionFactory(content__page__node__depth=2)
+        pagecontent = version.content
+        version.publish(self.get_superuser())
+        with self.login_user_context(self.get_superuser()):
+            response = self.client.post(
+                self.get_admin_url(PageContent, "set_home_content", pagecontent.pk)
+            )
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(
+            response.content.decode("utf-8"), "The page is not eligible to be home."
+        )
+
+    def test_non_existing_page_should_result_404(self):
+        with self.login_user_context(self.get_superuser()):
+            response = self.client.post(
+                self.get_admin_url(PageContent, "set_home_content", 99)
+            )
+        self.assertEqual(response.status_code, 404)
+
+    def test_permission_to_set_home_page(self):
+        version = PageVersionFactory(content__page__node__depth=1)
+        pagecontent = version.content
+        version.publish(self.get_superuser())
+
+        with self.login_user_context(self.get_superuser()), patch(
+            "djangocms_versioning.admin.VersioningAdminMixin.has_change_permission",
+            return_value=False,
+        ) as mock:
+            response = self.client.post(
+                self.get_admin_url(PageContent, "set_home_content", pagecontent.pk)
+            )
+        self.assertEqual(mock.call_count, 1)
+        self.assertEqual(response.status_code, 403)
+
```

Comment:
You're missing tests for the apphook signal firing stuff. You need one for when there are affected apphooks and the signal fires and one when there are no affected apphooks and the signal does not fire.